### DISCUSSION TITLE
Snippets in neovim terminal

### DISCUSSION THREAD

**Discussion by derekthecool** [L3MON4D3/LuaSnip#951] [General]
I'd love to use luasnippets in the neovim terminal. It would be awesome to have some superfast shell snippets for bash and powershell. There are some tools that help you run commands you've already run, but that is not enough.
With snippets in a neovim shell it would open the door to crazy fast terminal input.

I know there are likely to be several complications with shell readline features, new lines, and more.

I'm not even sure how to get started enabling this? I can see that the filetype of the new terminal (`tabnew | term`) is empty (no filetype is set). I can change the filetype like any other file, and the highlighting does change, but I don't get luasnip, cmp to start working.

Anybody have an idea about how to get started on this endeavor?

**Comment by eissar** [comment]
someone made a PR in the nvim-cmp repo for this. very interesting idea...
https://github.com/hrsh7th/nvim-cmp/pull/1805/commits/e47c774b953a10d3401a48eb02312b10e4466b28
jumping right and left between insert nodes is awesome with luasnip so i can see some really cool workflows with this.
you could kinda do something like document commands you don't use very often, and make the insert node a comment to yourself, e.g.:
```
get-todos -date <insert('date in format yyyy-mm-dd')>
```
on another note you can use the full-fat flag in the command name, so the command would be more explicit and you wouldn't need to remember tons of single letter flags. 

probably feasable workaround would be to have a keybind to show a command-bar like buffer that you can type your command into with normal snippet completion. Then just send the buftext as a command to a new/existing terminal

**Reply by derekthecool** [reply]
Wow, thanks for the feedback!
It makes sense it would have to be implemented at the CMP level. I'll have to give that a try!

I use stenography for my development so I have many tricks for easily triggering snippets and this could be a game changer for interactive shell usage.
